src/main/java/core/basesyntax/strategy/impl/TransactionServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.strategy.Operation;
+import core.basesyntax.strategy.TransactionService;
+import core.basesyntax.strategy.transactions.FruitTransaction;
+import core.basesyntax.strategy.transactions.TransactionHandler;
+import java.util.HashMap;
+import java.util.List;
+
+public class TransactionServiceImpl implements TransactionService {
+    private final HashMap<Operation, TransactionHandler> transactionMap;
+
+    public TransactionServiceImpl(HashMap<Operation, TransactionHandler> transactionMap) {
+        this.transactionMap = transactionMap;
+    }
+
+    public void applyTransactions(List<FruitTransaction> actions) {
+        actions.forEach(this::transactionStrategy);
+    }
+
+    private void transactionStrategy(FruitTransaction fruitTransaction) {
+        transactionMap.get(fruitTransaction.getOperation())
+                .produceTransaction(fruitTransaction);
+    }

```

Comment:
<img width="816" alt="image" src="https://user-images.githubusercontent.com/58032907/198897831-64e9b823-6b30-4009-8f7f-97603e727eec.png">

also `transactionStrategy` isn't best name for a method, (but good for a class) because methods do actions and should starts with a verb